lib/py/src/transport/TSocket.py

```diff
@@ -91,12 +91,13 @@ def isOpen(self):
             try:
                 peeked_bytes = self.handle.recv(1, socket.MSG_PEEK)
             except (socket.error, OSError) as exc:  # on modern python this is just BlockingIOError
-                if exc.errno == errno.EWOULDBLOCK:
+                if exc.errno in (errno.EWOULDBLOCK, errno.EAGAIN):
                     return True
-                raise
+                return False
```

Comment:
I think we also need to catch exceptions that's not `socket.error` nor `OSError` to return False, so maybe it's better to have a catch-all on the upper level of try...finally to just `return False` for all exceptions? I know catch-all is usually discouraged in python linters but am not sure if there's a better approach here.

Reply:
What kinds of errors are you concerned about? An over-broad catch like that would mask if someone made a legitimate bug in the covered code. AFAIK even in 2.7 `socket.error` should catch everything actually socket related: https://docs.python.org/2/library/socket.html#socket.error